File: src/main/java/com/dingdangmaoup/resin/pura/ResinBundle.java
/*
 * Copyright (c) 2005 JetBrains s.r.o. All Rights Reserved.
 */
package com.dingdangmaoup.resin.pura;

import com.intellij.DynamicBundle;
import java.util.function.Supplier;
import org.jetbrains.annotations.Nls;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.PropertyKey;

public final class ResinBundle {
  public static final @NonNls String BUNDLE = "messages.ResinBundle";
  private static final DynamicBundle INSTANCE = new DynamicBundle(ResinBundle.class, BUNDLE);

  private ResinBundle() {}

  public static @NotNull @Nls String message(@NotNull @PropertyKey(resourceBundle = BUNDLE) String key, Object @NotNull ... params) {
    return INSTANCE.getMessage(key, params);
  }

  public static @NotNull Supplier<@Nls String> messagePointer(@NotNull @PropertyKey(resourceBundle = BUNDLE) String key, Object @NotNull ... params) {
    return INSTANCE.getLazyMessage(key, params);
  }
}

File: src/main/kotlin/com/dingdangmaoup/resin/pura/ResinBundle.kt
package com.dingdangmaoup.resin.pura

import com.intellij.DynamicBundle
import java.util.function.Supplier
import org.jetbrains.annotations.Nls
import org.jetbrains.annotations.NonNls
import org.jetbrains.annotations.PropertyKey

object ResinBundle {
    @NonNls
    const val BUNDLE: String = "messages.ResinBundle"

    private val INSTANCE = DynamicBundle(ResinBundle::class.java, BUNDLE)

    @JvmStatic
    @Nls
    fun message(@PropertyKey(resourceBundle = BUNDLE) key: String, vararg params: Any): String {
        return INSTANCE.getMessage(key, *params)
    }

    @JvmStatic
    fun messagePointer(@PropertyKey(resourceBundle = BUNDLE) key: String, vararg params: Any): Supplier<@Nls String> {
        return INSTANCE.getLazyMessage(key, *params)
    }
}
